### DISCUSSION TITLE
VM not found by check/virsh

### DISCUSSION THREAD

**Discussion by Milliw** [winapps-org/winapps#28] [Support]
Hi!

At first: Thanks for that project!

Second:
I did everything as described and with GUI virtman I see my VM but running `bin/winapps check` it says, it is not running.
`virsh list --all` lists no vm, only when run with sudo.

I'm on Nobara 38 (based on Fedora 38).

**Comment by oskardotglobal** [comment]
Hello,
- Can you provide the exact output of `bin/winapps check`?
- Did you enable RDP for your VM and change the device name? 
- Are you running Wayland or X11?
- Did you run the installer?

Also, we're currently in the process of rewriting the entire project (we're not even the original authors, you can thank @fmstrat for creating the project) and now recommend using [`quickemu`](https://github.com/quickemu-project/quickemu), which allows you to create VMs with zero configuration.

**Comment by LDprg** [comment]
You need to run winapps as sudo, because each user has it own vms. The problem is that virt manager run on the root user per default nowadays. So winapps cannot access the vm if not also run as root. 

@oskardotglobal We need to improve and update the tutorial/doc. This problem with different users is something I also run into almost every time. However quickemu I think should solve the problem.

**Reply by oskardotglobal** [reply]
Yeah, I was thinking about the same thing. We probably need a rewrite of the docs, but before we do that I'd like to create the custom quickemu "image" and maybe even create our own script based on quickemu since we don't need the distro choose functionality. 
Also, if we do that, we need to change some stuff in the legacy install script since it deleted the vm on my machine (because it was in ~/.local/share/winapps)

**Reply by Milliw** [reply]
Hm yeah ok. But when I `sudo bin/winapps check` I get `You need to create a ~/.config/winapps/winapps.conf configuration. Exiting...` though that file already exists. I can't run the installer script neither.

EDIT:
Oh - and one question: Does it work on Wayland at all? ^^

**Reply by LDprg** [reply]
Yeah now the problem you are running into is that winapps will search for the config in the home directory of the root user. I normally just link the folder from my user to the root directory.

Winapps works very unstable currently on wayland (it uses XWayland). X11 is currently recommended and the only officially supported one. However we are working on a stable solution for wayland, but this will take some time.

**Comment by Milliw** [comment]
@LDprg Would you mind explaining me how to do that?
I logged in as root in terminal and cd to home folder. Then into .config and created winapps folder aswell as the config file but winapps still says it is not present.

**Reply by LDprg** [reply]
@Milliw here a short explanation how I did it.

- log into the root user with sudo -i or sudo su
- link the created winapps folder (not file) from the user to the root user using ln -s, here always use absolute paths to avoid wrong user directories 
- Double check your paths and file names with ls (I often ended with cascaded folders or wrong file names which does not work)
- Check the config file according to the doc
- Check vm with virsh (root user) 
- run the winapps command with sudo and if that doesn't help, login into the root user and try again

Feel free to ping me again whether it works or not.

**Reply by Milliw** [reply]
Thx for your help!

I managed to get it installed. I had to set groups for root user aswell and I also figured, it maybe would be a good idea to use the files from this repo and not the original one instead ;)

So far it seems to run quite well!
I tried some of the default Windows programs and they work - even under Wayland.
Sadly Adobe InDesign CS5-5 bugs out mostly all of the time being not usable and/or resizes to extremely small window. I managed to get it working by running msedge.exe after it but can not reproduce it always. Under X it runs well!

So yeah - I have to use an X session for that particular program I guess until Wayland sessions are better supported. But hey - at least I don't have to startup Windows anymore for this!! =)

EDIT:
Is there a way to close those "VM windows" reliably? Can't seem to find them in the processes (which is to be expected I guess) and if a program does not work I have to restart the VM so far.

**Reply by LDprg** [reply]
You can force quit it with virt-manager or use virsh shutdown <name>.

Also you might consider switching from virt-manager to quickemu. It is easier to use and setup

**Reply by Milliw** [reply]
That's what I already did if a program behaved weird.
But I also thought of starting the Windows task manager then. This way I can close down only that program. =) 

Yet I could not find out how to start the applications through the .desktop file with root permissions.
I tried several ways like editing the file and trying to use sudo as well as pkexec but I could not succeed. Researching on how to do this wasn't successful either and it appeared to me that this isn't that easy to do?!

I mean, I can start program from terminal no problem but beeing able to click that gui entries would be the preferred way to go.

**Comment by oskardotglobal** [comment]
Maybe try gksu or kdesudo
